### PR TITLE
[flutter_conductor] Rewrite writeStateToFile to be in an overidable method.

### PR DESCRIPTION
**Pull request by christopherfujino**
Part of https://github.com/flutter/flutter/issues/93156

### CODE REVIEW

File: dev/conductor/core/lib/src/next.dart [hunk 1]
```diff
@@ -149,7 +149,7 @@ class NextContext {
           );
           if (!response) {
             stdio.printError('Aborting command.');
-            writeStateToFile(stateFile, state, stdio.logs);
+            state_import.writeStateToFile(stateFile, state, stdio.logs);
```
Comment (Yugue) [inline dev/conductor/core/lib/src/next.dart:152]: Should this use `updateState(stateFile, state, stdio.logs)` as well?

Comment (christopherfujino) [reply dev/conductor/core/lib/src/next.dart:152]: good catch

Comment (christopherfujino) [reply dev/conductor/core/lib/src/next.dart]: updated, PTAL

**Review by Yugue — APPROVED** [review]

**Comment by Yugue** [comment]
LGTM

**Pull request merged**